pkg/supportbundle/spec.go

```diff
@@ -106,106 +106,200 @@ func CreateRenderedSpec(app apptypes.AppType, sequence int64, kotsKinds *kotsuti
+
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to inject defaults")
 	}
 
-	s := serializer.NewYAMLSerializer(serializer.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
-	var b bytes.Buffer
-	if err := s.Encode(builtBundle, &b); err != nil {
-		return nil, errors.Wrap(err, "failed to encode support bundle")
-	}
+	for key, builtBundle := range builtBundles {
+		s := serializer.NewYAMLSerializer(serializer.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
+		var b bytes.Buffer
+		if err := s.Encode(builtBundle, &b); err != nil {
+			return nil, errors.Wrap(err, "failed to encode support bundle")
+		}
 
-	templatedSpec := b.Bytes()
+		templatedSpec := b.Bytes()
 
-	renderedSpec, err := helper.RenderAppFile(app, &sequence, templatedSpec, kotsKinds, util.PodNamespace)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed render support bundle spec")
-	}
+		renderedSpec, err := helper.RenderAppFile(app, &sequence, templatedSpec, kotsKinds, util.PodNamespace)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed render support bundle spec")
+		}
 
-	// unmarshal the spec, look for image replacements in collectors and then remarshal
-	// we do this after template rendering to support templating and then replacement
-	supportBundle, err := kotsutil.LoadSupportBundleFromContents(renderedSpec)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to unmarshal rendered support bundle spec")
-	}
+		// unmarshal the spec, look for image replacements in collectors and then remarshal
+		// we do this after template rendering to support templating and then replacement
+		supportBundle, err := kotsutil.LoadSupportBundleFromContents(renderedSpec)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to unmarshal rendered support bundle spec")
+		}
+
+		var registrySettings registrytypes.RegistrySettings
+		if !util.IsHelmManaged() {
+			s, err := store.GetStore().GetRegistryDetailsForApp(app.GetID())
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to get registry settings for app")
+			}
+			registrySettings = s
+		}
 
-	var registrySettings registrytypes.RegistrySettings
-	if !util.IsHelmManaged() {
-		s, err := store.GetStore().GetRegistryDetailsForApp(app.GetID())
+		collectors, err := registry.UpdateCollectorSpecsWithRegistryData(supportBundle.Spec.Collectors, registrySettings, kotsKinds.Installation.Spec.KnownImages, kotsKinds.License, &kotsKinds.KotsApplication)
 		if err != nil {
-			return nil, errors.Wrap(err, "failed to get registry settings for app")
+			return nil, errors.Wrap(err, "failed to update collectors")
+		}
+		supportBundle.Spec.Collectors = collectors
+		b.Reset()
+		if err := s.Encode(supportBundle, &b); err != nil {
+			return nil, errors.Wrap(err, "failed to encode support bundle")
+		}
+		renderedSpec = b.Bytes()
+		secretName := GetSpecSecretName(app.GetSlug())
+		if key != "kotsadm" {
+			secretName = GetSpecSecretName(app.GetSlug() + "-" + key)
```

Comment:
We might want to come up with a different naming scheme since there is a possibility (though likely a very low one) that there could be a naming conflict.  Example, a user has two apps with slugs `my-app` and `my-app-vendor`.

Maybe we can name the new split specs differently to avoid this?
```
kotsadm-<appslug>-supportbundle-cluster-specific
kotsadm-<appslug>-supportbundle-default
kotsadm-<appslug>-supportbundle-vendor
```

Reply:
good idea, I have changed the name to be as your suggested, and updated 
```
appNameRE = regexp.MustCompile(`^kotsadm-.*-supportbundle(?:$|.*)`)
```
to filter out all kotsadm objects that have an object name